src/openads/application/service/AddPositionUseCase.js

```diff
@@ -28,12 +32,29 @@ export default class AddPositionUseCase {
     return this._positionRepository.find({id})
       .then(this._filterPositionAlreadyExists)
       .then(() => this._positionFactory.create({id, name, source, placement, segmentation, sizes, native}))
-      .then(position => this._positionRepository.saveOrUpdate({position}))
+      // .then(this._setAdToPosition) // todo why this line fails ¿? the line below is working fine insted
+      .then(createdPosition => this._setAdToPosition(createdPosition))
```

Comment:
This is javascript faults, Promise is translated to es5 like in a global scope of window and doesnt recognize "this".
A possible solution is just bind this to the method like this:
```.then(this._setAdToPosition.bind(this))```

Reply:
IMO the .bind(this) is a mindblower :) 
for readability I prefer, the arrow function option, is it OK for you?

Reply:
Yes no problem, but we agree on mix solutions? Some functions are pure and aren't using this, in that case we can use functions with one param like ```filterPositionAdIsAvailable```

Reply:
👍 Agree